tilecloud_chain/controller.py

```diff
@@ -387,15 +388,24 @@ def _generate_apache_config(gene):
                 dim = len(layer['dimensions'])
                 for r in res:
                     f.write(
-                        'RewriteRule ^%(tiles_location)s/1.0.0/%(layer)s/([a-zA-Z0-9_\-~\.]+)/([a-zA-Z0-9_\-~\.]+)/'
-                        '%(dimensions_re)s%(zoom)s/(.*)$ %(mapcache_location)s/wmts/1.0.0/%(layer)s/$1/$2/'
-                        '%(dimensions_rep)s%(zoom)s/%(final)s [PT]\n' % {
+                        'RewriteRule'
+                        ' '
+                        '^%(tiles_location)s/1.0.0/%(layer)s/%(token_regex)s/' # Baseurl/layer/Style
+                        '%(dimensions_re)s/'  # Dimensions : variable number of values
+                        '/%(token_regex)s/%(zoom)s/(.*)$' # TileMatrixSet/TileMatrix/TileRow/TileCol.extension
+                        ' '
+                        '%(mapcache_location)s/wmts/1.0.0/%(layer)s/$1'
+                        '/%(dimensions_rep)s'
+                        '/$%(dim+2)s/%(zoom)s/$%(dim+3)s'
+                        ' '
+                        '[PT]\n' % {
                             'tiles_location': gene.config['apache']['location'],
                             'mapcache_location': gene.config['mapcache']['location'],
                             'layer': layer['name'],
-                            'dimensions_re': ''.join(['([a-zA-Z0-9_\-~\.]+)/' for e in range(dim)]),
-                            'dimensions_rep': ''.join(['$%i/' % (e + 3) for e in range(dim)]),
-                            'final': '$%i' % (3 + dim),
+                            'param_regex': token_regex,
+                            'dimensions_re': '/'.join([token_regex for e in range(dim)]),
```

Comment:
You moved the '/' position than it will no more works without any dimensions ...